code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,842 @@
+	name = "Guise"
+	desc = "Wraps your form in shadows, making you harder to see."
+	panel = "Thrall Abilities"
+	charge_max = 1200
+	human_req = 1
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+	clothes_req = 0
+	action_icon_state = "shadow_walk"
+
+/obj/effect/proc_holder/spell/self/lesser_shadow_walk/proc/reappear(mob/living/carbon/human/user)
+	user.visible_message("<span class='warning'>[user] appears from nowhere!</span>", "<span class='shadowling'>Your shadowy guise slips away.</span>")
+	user.alpha = initial(user.alpha)
+
+/obj/effect/proc_holder/spell/self/lesser_shadow_walk/cast(mob/living/carbon/human/user)
+	user.visible_message("<span class='warning'>[user] suddenly fades away!</span>", "<span class='shadowling'>You veil yourself in darkness, making you harder to see.</span>")
+	user.alpha = 10
+	addtimer(CALLBACK(src, .proc/reappear, user), 40)
+
+
+
+/obj/effect/proc_holder/spell/targeted/night_vision/thrall //Toggleable night vision for thralls
+	name = "Thrall Darksight"
+	desc = "Allows you to see in the dark!"
+	action_icon_state = "darksight"
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+
+/obj/effect/proc_holder/spell/self/lesser_shadowling_hivemind //Lets a thrall talk with their allies
+	name = "Lesser Commune"
+	desc = "Allows you to silently communicate with all other shadowlings and thralls."
+	panel = "Thrall Abilities"
+	charge_max = 50
+	human_req = 1
+	clothes_req = 0
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+	action_icon_state = "commune"
+
+/obj/effect/proc_holder/spell/self/lesser_shadowling_hivemind/cast(mob/living/carbon/human/user)
+	if(!is_shadow_or_thrall(user))
+		to_chat(user, "<span class='warning'><b>As you attempt to commune with the others, an agonizing spike of pain drives itself into your head!</b></span>")
+		user.apply_damage(10, BRUTE, "head")
+		return
+	var/text = stripped_input(user, "What do you want to say your masters and fellow thralls?.", "Lesser Commune", "")
+	if(!text)
+		return
+	text = "<span class='shadowling'><b>\[Thrall\]</b><i> [user.real_name]</i>: [text]</span>"
+	for(var/mob/M in GLOB.mob_list)
+		if(is_shadow_or_thrall(M))
+			to_chat(M, text)
+		if(isobserver(M))
+			to_chat(M, "<a href='?src=\ref[M];follow=\ref[user]'>(F)</a> [text]")
+	log_say("[user.real_name]/[user.key] : [text]")
+
+
+// ASCENDANT ABILITIES BEYOND THIS POINT //
+// YES THEY'RE OP, BUT THEY'VE WON AT THE POINT WHERE THEY HAVE THIS, SO WHATEVER. //
+
+
+/obj/effect/proc_holder/spell/targeted/annihilate //Gibs someone instantly.
+	name = "Annihilate"
+	desc = "Gibs someone instantly."
+	panel = "Ascendant"
+	range = 7
+	charge_max = 0
+	clothes_req = 0
+	action_icon_state = "annihilate"
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+	sound = 'sound/magic/Staff_Chaos.ogg'
+
+/obj/effect/proc_holder/spell/targeted/annihilate/cast(list/targets,mob/living/simple_animal/ascendant_shadowling/user = usr)
+	if(user.incorporeal_move)
```

Comment:
this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.